tests/test_supp_1_channel_only.py

```diff
@@ -1,310 +1,312 @@
-#     # Hydro
-#     hydro_rst = candidate_channel_only_run_expected.restart_hydro[0]
-#     new_hydro_rst_path = run_dir.joinpath(hydro_rst.name)
-#     new_hydro_rst_path.unlink()
-#     new_hydro_rst_path.symlink_to(hydro_rst)
-#
-#     perfrestart_setup.hydro_namelist['hydro_nlist'].update(
-#         {'restart_file': str(new_hydro_rst_path)})
-#
-#     # Nudging
-#     if candidate_channel_only_run_expected.restart_nudging is not None and \
-#        len(candidate_channel_only_run_expected.restart_nudging) > 0:
-#         nudging_rst = candidate_channel_only_run_expected.restart_nudging[0]
-#         new_nudging_rst_path = run_dir.joinpath(nudging_rst.name)
-#         new_nudging_rst_path.unlink()
-#         new_nudging_rst_path.symlink_to(nudging_rst)
-#
-#         perfrestart_setup.hydro_namelist['nudging_nlist'].update(
-#             {'nudginglastobsfile': str(run_dir.joinpath(nudging_rst.name))})
-#
-#
-#     # Setup the restart in the run.
-#     orig_start_time, orig_end_time = wrfhydropy.job_tools.solve_model_start_end_times(
-#         None,
-#         None,
-#         candidate_channel_only_perfrestart_run.setup
-#     )
-#
-#     restart_dt = hydro_rst.open()
-#     restart_time = dt.datetime.strptime(restart_dt.Restart_Time,'%Y-%m-%d_%H:%M:%S')
-#
-#     candidate_channel_only_perfrestart_job.model_start_time = restart_time
-#     candidate_channel_only_perfrestart_job.model_end_time = orig_end_time
-#
-#     # Run
-#     with warnings.catch_warnings():
-#         warnings.simplefilter("ignore")
-#         candidate_channel_only_perfrestart_run.add_jobs(candidate_channel_only_perfrestart_job)
-#         check_run_dir = candidate_channel_only_perfrestart_run.run_jobs()
-#         if scheduler is not None:
-#             candidate_channel_only_perfrestart_run = \
-#                 wrfhydropy.job_tools.restore_completed_scheduled_job(check_run_dir)
-#
-#     #Check against initial run
-#     perfstart_restart_diffs = wrfhydropy.RestartDiffs(
-#         candidate_channel_only_perfrestart_run,
-#         candidate_channel_only_run_expected
-#     )
-#     ## Check hydro restarts
-#     for diff in perfstart_restart_diffs.hydro:
-#         if diff is not None:
-#             with capsys.disabled():
-#                 print(diff)
-#         assert diff is None, \
-#             "candidate_channel_only hydro restart files do not match when starting from a restart"
-#
-#     ## Check nudging restarts
-#     for diff in perfstart_restart_diffs.nudging:
-#         if diff is not None:
-#             with capsys.disabled():
-#                 print(diff)
-#         assert diff is None, \
-#             "candidate_channel_only nudging restart files do not match when starting from a restart"
\ No newline at end of file
+import copy
+import datetime as dt
+import os
+import pandas as pd
+import pathlib
+import pickle
+import time
+import warnings
+import wrfhydropy
+
+
+# Utility function to wait for job completion
```

Comment:
You can import this from the test_1_fundamental module

Alternatively, we can start another module for these type of functions